docs/storage.md

```diff
@@ -99,6 +100,8 @@ config:
 
 At a minimum, you will need to provide a value for the `bucket`, `endpoint`, `access_key`, and `secret_key` keys. The rest of the keys are optional.
 
+However if you set `aws_sdk_auth: true` Thanos will use the default authentication methods of the AWS SDK for go based on [known environment variables](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html) (`AWS_PROFILE`, `AWS_WEB_IDENTITY_TOKEN_FILE` ... etc) and known AWS config files (~/.aws/config). If you turn this on, then the `bucket` and `endpoint` are the required config keys.
```

Comment:
Last comment. This says:
> If you turn this on, then the `bucket` and `endpoint` are the required config keys.

We have this function: https://github.com/thanos-io/thanos/blob/6b6a39f2a6073c44a434728dc9e958b1da333dd7/pkg/objstore/s3/s3.go#L343-L366

Do we need to make any adjustments to it? Because it seems like this sentence doesn't match up with that code :thinking:

Reply:
I guess we can move https://github.com/thanos-io/thanos/pull/4667/files#diff-03a36882ad7e01567144dad0fb3742a57a0c84a9d2dbc3c3077e941d2e7ff67cR229-R231 in validate ?

Reply:
Done.